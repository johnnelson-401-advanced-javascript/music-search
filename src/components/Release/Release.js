import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default function Release({ title, id, imageUrl, artist }) {
  return (
    <Link to={`/songs/${id}/${artist}`}>
      <h3>{title}</h3>
      <img src={imageUrl} alt={title} />
    </Link>
  
  );
}

Release.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  artist: PropTypes.string.isRequired
};
