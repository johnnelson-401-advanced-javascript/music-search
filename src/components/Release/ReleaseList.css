.ReleaseList {
 
  display: flex;
  flex-wrap: wrap;
  border: 3px solid;
  li{
    border: 1px solid black;
    margin: 3px;
    padding: 5px;
    width: 200px;
    height: 200px;
    list-style: none;
    img {
      height: 75%;
      margin-top: 10px;
      margin-left: 20px;
    }
    h3 {
      text-align: center;
      margin-top: 8px;
    }

  }


}