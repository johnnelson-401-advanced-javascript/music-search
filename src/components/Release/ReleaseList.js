import React from 'react';
import PropTypes from 'prop-types';
import Release from './Release';
import styles from './ReleaseList.css';

export default function ReleaseList({ releases, artist }) {

  const releaseElements = releases.map(release => (
    <li key={release.id}>
      <Release artist={artist} title={release.title} id={release.id} imageUrl={release.imageUrl}/>
    </li>
  ));

  return (
    <ul className={styles.ReleaseList}>
      {releaseElements}
    </ul>
  );
}
ReleaseList.propTypes = {
  releases: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired
  })).isRequired,
  artist: PropTypes.string.isRequired,

};
