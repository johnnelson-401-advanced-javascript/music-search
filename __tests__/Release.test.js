import React from 'react';
import { shallow } from 'enzyme';
import Release from '../src/components/Release/Release';

describe('Release component', () => {
  it('renders Release', () => {
    const wrapper = shallow(<Release title='Phil' imageUrl='testUrl.com' id='id' artist='artist' />);
    expect(wrapper).toMatchSnapshot();
  });
});
