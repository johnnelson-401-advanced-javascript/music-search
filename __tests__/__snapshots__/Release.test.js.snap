// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders Release 1`] = `
<Link
  to="/songs/id/artist"
>
  <h3>
    Phil
  </h3>
  <img
    alt="Phil"
    src="testUrl.com"
  />
</Link>
`;
